bin/propolis-server/src/lib/vm/ensure.rs

```diff
@@ -226,7 +226,26 @@ impl<'a> VmEnsureNotStarted<'a> {
                     kernel_vm_paused: false,
                 })
             }
-            Err(e) => Err(self.fail(e).await),
+            Err(e) => {
```

Comment:
One minor caveat: this works in part because in today's code, no one ever actually drops a `VmEnsureObjectsCreated` (both code paths that instantiate one more or less immediately call `ensure_active` on it, which infallibly moves the runtime to `Vm::make_active`; after that point it's guaranteed that someone will call `shutdown_background` on the runtime before it gets dropped).

I think fixing this might be mildly involved (though maybe there's an elegant solution I haven't seen yet), so I don't think I'd hold up the whole PR for it, but it might be good to note this property in a comment somewhere around here.

Reply:
so, from this nudging and our chat, i think being more intentional about creating and initializing tasks on `vmm_rt` is pretty workable! that's how i've got the PR now. at least this leaves us with a clearly-safe way to handle fallibility on the path to a `VmEnsureObjectsCreated`.

if we decided to do something fallible between a `VmEnsureObjectsCreated` and an `ensure_active`, well, that would be pretty unfortunate.